components/calendar/calendar.js

```diff
@@ -326,53 +334,58 @@ class Calendar extends LocalizeStaticMixin(LitElement) {
 		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear);
 		const dayRows = dates.map((week) => {
 			const weekHtml = week.map((day) => {
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				const selected = this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false;
 				const classes = {
-					'd2l-calendar-date': true,
-					'd2l-calendar-date-selected': this.selectedValue ? checkIfDatesEqual(day, parseISODate(this.selectedValue)) : false,
+					'd2l-calendar-date-selected': selected,
 					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
 				};
-				const focused = checkIfDatesEqual(day, this._focusDate);
+				const year = day.getFullYear();
+				const month = day.getMonth();
+				const date = day.getDate();
 				return html`
-					<td>
-						<div
-							@click="${this._onDateSelected}"
-							@keydown="${this._onKeyDown}"
-							class=${classMap(classes)}
-							data-date=${day.getDate()}
-							data-month=${day.getMonth()}
-							data-year=${day.getFullYear()}
-							tabindex=${focused ? '0' : '-1'}>
-							${day.getDate()}
-						</div>
+					<td
+						aria-label="${formatDate(day, {format: 'medium'})}"
```

Comment:
I used medium instead of full because the column header information adds the context of the weekday name